File: Math/CPP/minimum_cost_to_move_chips_to_the_same_position.cpp
#include <algorithm>
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    int minCostToMoveChips(vector<int>& position) {
        int odd = 0, even = 0;
        unordered_map<int, int> counter;
        for (const int n: position) counter[n]++;
        for (const auto [k, v] : counter) {
            if (k % 2 == 0) even += v;
            else odd += v;
        }
        return min(odd, even);
    }
};

File: Math/Python/minimum_cost_to_move_chips_to_the_same_position.py
from collections import Counter

class Solution:
    def minCostToMoveChips(self, position: list[int]) -> int:
        position = Counter(position)
        even, odd = 0, 0
        for k,v in (position.items()):
            if k & 1: odd += v
            else: even += v
        return min(even, odd)